File: ptm_schedule/src/kernel/properties/keycloak_properties.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package properties

import "github.com/golibs-starter/golib/config"

type KeycloakProperties struct {
	Url              string
	JwkSetUri        string
	Realm            string
	ClientId         string
	ClientSecret     string
	ExpectedIssuer   string
	ExpectedAudience string
}

func (k KeycloakProperties) Prefix() string {
	return "app.keycloak"
}

func NewKeycloakProperties(loader config.Loader) (*KeycloakProperties, error) {
	var props KeycloakProperties
	if err := loader.Bind(&props); err != nil {
		return nil, err
	}
	return &props, nil
}

File: crm/src/main/java/serp/project/crm/kernel/property/KeycloakProperties.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.crm.kernel.property;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "app.keycloak")
@Data
public class KeycloakProperties {
    private String url;
    private String jwkSetUri;
    private String realm;
    private String clientId;
    private String clientSecret;
    private String expectedIssuer;
    private String expectedAudience;
}
